gbpservice/nfp/configurator/agents/loadbalancer_v2.py

```diff
@@ -0,0 +1,979 @@
+import os
+
+from oslo_log import log as logging
+
+from gbpservice.nfp.common import exceptions
+from gbpservice.nfp.configurator.agents import agent_base
+from gbpservice.nfp.configurator.lib import data_filter
+from gbpservice.nfp.configurator.lib import lbv2_constants as lb_constants
+from gbpservice.nfp.configurator.lib import utils
+from gbpservice.nfp.core import event as nfp_event
+from gbpservice.nfp.core import poll as nfp_poll
+
+LOG = logging.getLogger(__name__)
+
+""" Implements LBaaS response path to Neutron plugin.
+Methods of this class are invoked by the LBaaSV2EventHandler class and also
+by driver class for sending response from driver to the LBaaS Neutron plugin.
+"""
+
+
+class LBaaSV2RpcSender(data_filter.Filter):
+
+    def __init__(self, sc):
+        self.notify = agent_base.AgentBaseNotification(sc)
+
+    # TODO(jiahao): copy from v1 agent need to review
+    def get_logical_device(self, pool_id, context):
+        """ Calls data filter library to get logical device from pool_id.
+
+        :param pool_id: object type
+        :param context: context which has list of all pool related resources
+                        belonging to that tenant
+
+        Returns: logical_device
+        """
+        return self.call(
+            context,
+            self.make_msg(
+                'get_logical_device',
+                pool_id=pool_id
+            )
+        )
+
+    def update_status(self, obj_type, obj_id, root_lb_id,
+                      provisioning_status, operating_status,
+                      agent_info, obj=None):
+        """ Enqueues the response from LBaaS V2 operation to neutron plugin.
+
+        :param obj_type: object type
+        :param obj_id: object id
+        :param root_lb_id: root loadbalancer id
+        :param provisioning_status: an enum of (‘ACTIVE’, ‘PENDING_CREATE’,
+        ‘PENDING_UPDATE’, ‘PENDING_DELETE’, ‘ERROR’)
+        :param operating_status:  an enum of
+        (‘ONLINE’, ‘OFFLINE’, ‘DEGRADED’, ‘ERROR’)
+
+        """
+
+        msg = {'info': {'service_type': lb_constants.SERVICE_TYPE,
+                        'context': agent_info['context']},
+               'notification': [{'resource': agent_info['resource'],
+                                 'data':{'obj_type': obj_type,
+                                         'obj_id': obj_id,
+                                         'notification_type': 'update_status',
+                                         'root_lb_id': root_lb_id,
+                                         'provisioning_status':
+                                             provisioning_status,
+                                         'operating_status':
+                                             operating_status,
+                                         obj_type: obj}}]
+               }
+        self.notify._notification(msg)
+
+    # TODO(jiahao): copy from v1 agent need to review
+    def update_pool_stats(self, pool_id, stats, context, pool=None):
```

Comment:
This is used in the code to get stats and we are posting. Can you check if the events are stats are getting updated in the loadbalancer Neutron resource?